### PR TITLE
`quantize_linear` compositional func in Experimental API

### LINKED ISSUE #16427: QuantizeLinear

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #16427

### CODE REVIEW

**Comment by mobley-trent** [comment]
Temporarily removed support for the `axis` argument. Having issues with the tests

**Comment by Aarsh2001** [comment]
Commenting for context, PR is related to this task -: https://trello.com/c/bM30qW9Z

**Comment by Aarsh2001** [comment]
Hey @mobley-trent, kindly let us know in the comments whenever this PR is ready for review, thanks 🙂

**Comment by mobley-trent** [comment]
@Aarsh2001 I don't know why JAX is returning inconsistent results with the rest of the backends

**Comment by mobley-trent** [comment]
Just pushed a fix for the inconsistent test results, but I still don't know how we should proceed to support the `axis` argument

File: ivy/functional/ivy/experimental/manipulation.py [hunk 1]
```diff
@@ -30,6 +30,53 @@
...
+    >>> scale = 0.1
+    >>> zero_point = 128
+    >>> axis = 1
+    >>> ivy.quantize_linear(x, scale, zero_point, axis=axis)
+    ivy.array([[143., 155.], [136., 160.]])
+    """
+    y = ivy.round(x / scale + zero_point).astype(ivy.uint8)
+
+    if saturate:
+        y = ivy.clip(y, 0, 255)
```
Comment (MahmoudAshraf97) [inline ivy/functional/ivy/experimental/manipulation.py:75]: based on onnx docs, this function should be generala and not constrained to certain output types or ranges, so I suggest making it more general and remove the hardcoded dtypes and limits

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/experimental/manipulation.py (hunk 1 above)
Comment (mobley-trent) [reply ivy/functional/ivy/experimental/manipulation.py:76]: There are a couple of specifications when it comes to the dtypes.

Comment (MahmoudAshraf97) [reply ivy/functional/ivy/experimental/manipulation.py:76]: what is mentioned here is that saturation is dependant on the input dtype, but not strictly restricted to `int8` and `uint8`

**Comment by mobley-trent** [comment]
The code is running fine but I'm still running into problems setting up the test

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy/functional/ivy/experimental/manipulation.py [hunk 2]
```diff
@@ -31,6 +31,93 @@
...
+        raise ValueError("Input 'x' must be of type float or int32.")
+    if y_zero_point.dtype not in [ivy.int8, ivy.uint8]:
+        raise ValueError("'y_zero_point' must be of type int8 or uint8.")
+
+    if y_scale.shape != y_zero_point.shape:
+        raise ValueError("y_scale and y_zero_point must have the same shape.")
+
+    if axis is not None:
+        if axis < 0:
+            axis = x.ndim + axis
```
Comment (MahmoudAshraf97) [inline ivy/functional/ivy/experimental/manipulation.py:100]:
```suggestion
```
not needed

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

**Comment by MahmoudAshraf97** [comment]
Hi @mobley-trent , are you still working on this function?

**Comment by mobley-trent** [comment]
@MahmoudAshraf97 I put it on hold since there doesn't seem to be active development on the issue list on these functions, as I have not seen any other PRs being merged. But my PR is still active.

**Comment by mobley-trent** [comment]
My tests are not running for some reason. Even the function in the Ivy API doesn't work when I try running it on a sample script; I'm getting this error:

```
ImportError: cannot import name '_wrap_function' from partially initialized module 'ivy.func_wrapper' (most likely due to a circular import) (/workspaces/ivy/ivy/func_wrapper.py)
```

**Comment by MahmoudAshraf97** [comment]
hi @mobley-trent , I'm going to close this PR to clear some PR backlog, feel free to open a new one once this function is ready to merge

**Pull request closed without merging**